README.md

```diff
@@ -0,0 +1,130 @@
+# Tahoe-LAFS Infrastructure
+
+The software that defines the Tahoe-LAFS project's infrastructure.
+
+## Abstract
+
+This repository was initially created in Sep 2022 as an effort to deploy and
+manage a `testgrid` server using NixOS (now retired).
+
+Since January 2025, some members of the community are seeking to re-use this repository
+to deploy and manage some new systems:
+
+- a `testgrid` server re-using the NixOS configuration written in 2022
+- a `webforge` server to [MoveOffTrac](https://github.com/tahoe-lafs/MoveOffTrac)
+
+In addition to the [Nix](https://nixos.org/) code allowing to manage the software installed on those servers,
+an [OpenToFu]() plan should handle the definition of the related infrastructure (e.g.: VMs and DNS records).
+
+## ToDo/WiP
+
+The content of this repository is mostly a work in progress (as time of writing).
+
+Here follows a list of things to do, if not already in progress (see open issues).
+
+For [Testgrid](#testgrid):
+
+- [X] Fix and rework the Nix code to deploy the current Tahoe-LAFS software
+- [ ] Implement or adapt the OpenToFu plan to cover the `testgrid` and
+  import the resource already deployed (on Hetzner and Gandi if possible)
+- [ ] Refactor the Nix code to integrate the `nixosConfiguration` with the top-level flake,
+  which will be used to manage and automated the deployment of all NixOS configurations
+
+For [Webforge](#webforge):
+
+- [ ] Implement the OpenToFu plan for Hetzner and its CI/CD workflow with GHA
+- [ ] Implement the NixOS configuration and its CI/CD workflow with GHA
+
+For the [Deployment Workflow](#deployment-workflow):
+
+- [ ] Automated the DNS configuration using OpenToFu and the relevant credentials from Gandi
+- [ ] Describe the [provisioning](#provisioning) steps with the relevant paths and snippets
+- [ ] Investigate how to allow pull request from (external) fork (with read-only token?)
+
+## How to Contribute
+
+- Read the doc and the code, consult the existing issues and ask questions (via `#tahoe-lafs` on [Libera](https://libera.chat/))
+- Describe the bug/problem or the missing info/feature in a new issue in this repository
+- Create a new branch (requires **Write** permissions in this repository) and start hacking
+- Submit pull requests (preferably one per issue) and verify the check status
```

Comment:
So you're making people create branches here (which requires elevated permissions) so they can benefit from the CI checks?

I think they shouldn't need to get permissions to collaborate here.  Either someone already privileges, this could make it easier for them to also collaborate here;  Or they create a PR as an outsider, having either not tested their changes locally, with a VM or not at all (and put that on us).

Reply:
I agree with you @hacklschorsch: it would be better to not need permissions to collaborate.

But you've also raised a valid concern in my earlier test where I was planning to make the ToFu state public.

After some thinking, I've found it rather wise to avoid - at least for a start - but the direct consequence is that only local branch will be able to access the state (because it requires a secret, same as for the 3rd party API like Hetzner). 

This is why I'm proposing to make a first step for which the Write permission requirement is the price to pay to keep the ToFu state private.
After what we should try to figure how to improve (described in the Roadmap section/page):

> Investigate how to allow pull request from (external) fork (with read-only token?)

Reply:
@hacklschorsch, I've read the section and your comments once more.

If I understand correctly, your concern is centered on Nix while my constraint comes (mostly) from OpenToFu.

So I came up with 106c0d2 which is an attempt to deal with this permission issue in a softer way.

I hope this help, though I'm not sure how we - as maintainer - will be dealing with this.
Do you see the maintainers replicating external branches locally and duplicate the pull request?

Reply:
Ah OK, I hadn't recalled that the TF "plan" phase already need the real secrets.  Hm.

I'd still value people opening PRs - even if they can't test/deploy to the tahoe-lafs infra and us maintainers have to "go the last mile", i.e. maybe by cloning their branches, or checking our the code and running the tests.

Reply:
Maybe I come up with something until tomorrow.
Any way, let's not require commit access to the repo for improvement proposals if we can avoid it?

Reply:
It is no longer required from the README.
I'll have to tune #21 to skip the ToFu check for external PR, but it should not be a problem.